### PR TITLE
BaSP-Feature-TG-Super Admin

### PR DESCRIPTION
**Pull request by FiolAxel**
This is the SuperAdmin branch where the following items could be created :

1. Get a SuperAdmin.
2. Get all SuperAdmins.
3. Post a new SuperAdmin.
4. Delete a SuperADmin.
5. Update.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,6 +25,7 @@
     "node": "14.x"
   },
   "dependencies": {
+    "body-parser": "^1.20.0",
```
Comment (cocolobos) [inline package.json:28]: we should not install new dependencies

File: src/data/super-admins.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    {"id":3,"firstName":"Lyman","lastName":"Adenet","email":"ladenet2@mit.edu","password":"Y7cRjBZm","active":true},
+    {"id":4,"firstName":"Nerti","lastName":"Smithe","email":"nsmithe3@jiathis.com","password":"H8fjmXf3F4nk","active":false},
+    {"id":5,"firstName":"Lil","lastName":"Ligertwood","email":"lligertwood4@netvibes.com","password":"ZspZFF81f3e","active":false},
+    {"id":6,"firstName":"Murvyn","lastName":"Tomkowicz","email":"mtomkowicz5@digg.com","password":"QDWshb","active":true},
+    {"id":7,"firstName":"Rois","lastName":"Wittman","email":"rwittman6@buzzfeed.com","password":"P5NVT3g5O","active":true},
+    {"id":8,"firstName":"Clarke","lastName":"Shout","email":"cshout7@cnbc.com","password":"k1HUEzT6","active":false},
+    {"id":9,"firstName":"Maurise","lastName":"Clausson","email":"mclausson8@lulu.com","password":"Yh6eUYx","active":true},
+    {"id":10,"firstName":"Kevan","lastName":"Meade","email":"kmeade9@nymag.com","password":"jkp1oE1","active":true},
+    {"id":11,"firstName":"Eleen","lastName":"Emmett","email":"eemmett5j@technorati.com","password":"bMtZxF21D","active":true}]
```
Comment (cocolobos) [inline src/data/super-admins.json:11]: can we add JSON format ?

File: src/index.js [hunk 3]
```diff
@@ -1,23 +1,23 @@
 // use "import" to import libraries
-import express from 'express';
+// import express from 'express';
```
Comment (cocolobos) [inline src/index.js:2]: do we need this change ?

File: src/index.js [hunk 4]
```diff
@@ -1,23 +1,23 @@
...
 const app = express();
 const port = process.env.PORT || 3000;
-
+const bodyParser = require('body-parser');
+const superAdmins = require('./resources/super-admins');
+// configure the app to use bodyParser()
+app.use(bodyParser.urlencoded({
+  extended: true,
+}));
+app.use(bodyParser.json());
```
Comment (cocolobos) [inline src/index.js:14]: app.use(express.json()) this is the same that body parser

File: src/resources/super-admins.js [hunk 5]
```diff
@@ -0,0 +1,110 @@
+const express = require('express');
+
+const router = express.Router();
+const fs = require('fs');
+const superAdmins = require('../data/super-admins.json');
+
+router.use((req, res, next) => {
+  next();
+});
```
Comment (lucasrendo) [inline src/resources/super-admins.js:9]: is this necessary?

File: src/resources/super-admins.js [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+router.post('/', (req, res) => {
+  const newSuperAdmin = {
+    id: superAdmins.length + 1,
```
Comment (lucasrendo) [inline src/resources/super-admins.js:13]: If i have 10 superAdmins and delete the first one, the last one would have the id: 10 and the length of the array would be 9 so newSuperAdmin would also have an id of 10. So this id generation can generate duplicated ids

**Review by lucasrendo — APPROVED** [review]

File: src/data/super-admins.json [hunk 7]
```diff
@@ -0,0 +1 @@
...:1,"firstName":"axelito","lastName":"bala 0016","phone":"341567897","email":"carlosbala75@hotmail.com","password":"1997323","active":true},{"id":2,"firstName":"Lyman","lastName":"Adenet","email":"ladenet2@mit.edu","password":"Y7cRjBZm","active":true},{"id":3,"firstName":"Nerti","lastName":"Smithe","email":"nsmithe3@jiathis.com","password":"H8fjmXf3F4nk","active":false},{"id":4,"firstName":"Lil","lastName":"Ligertwood","email":"lligertwood4@netvibes.com","password":"ZspZFF81f3e","active":false},{"id":5,"firstName":"Murvyn","lastName":"Tomkowicz","email":"mtomkowicz5@digg.com","password":"QDWshb","active":true},{"id":6,"firstName":"Rois","lastName":"Wittman","email":"rwittman6@buzzfeed.com","password":"P5NVT3g5O","active":true},{"id":7,"firstName":"Clarke","lastName":"Shout","email":"cshout7@cnbc.com","password":"k1HUEzT6","active":false},{"id":8,"firstName":"Maurise","lastName":"Clausson","email":"mclausson8@lulu.com","password":"Yh6eUYx","active":true},{"id":9,"firstName":"Kevan","lastName":"Meade","email":"kmeade9@nymag.com","password":"jkp1oE1","active":true},{"id":10,"firstName":"Eleen","lastName":"Emmett","email":"eemmett5j@technorati.com","password":"bMtZxF21D","active":true}]
```
Comment (cocolobos) [inline src/data/super-admins.json:1]: JSON file needs to be format

File: src/index.js [hunk 8]
```diff
@@ -1,11 +1,13 @@
 // use "import" to import libraries
 import express from 'express';
 // use "require" to import JSON files
+const superAdmins = require('./resources/super-admins');
 const admins = require('./data/admins.json');
 const projects = require('./resources/projects');
-
+// configure the app to use bodyParser()
```
Comment (cocolobos) [inline src/index.js:7]: remove comment

**Pull request merged**